src/coreclr/tools/Common/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -55,6 +55,10 @@ protected override ModuleData CreateValueFromKey(EcmaModule key)
             }
         }
         private readonly ModuleHashtable _moduleHashtable = new ModuleHashtable();
+        internal LockFreeReaderHashtable<EcmaModule, ModuleData>.Enumerator ModuleEnumerator()
```

Comment:
I don't think we need a new method for this. InheritOpenModules can just inline this: `foreach (ModuleData oldModuleData in ModuleHashtable.Enumerator.Get(oldTypeSystemContext._moduleHashtable))`